.github/workflows/cd.yaml

```
@@ -57,58 +43,57 @@ jobs:
       - run: |
           set -x
           NAMESPACE=ci-${{ github.run_number }}
+          SPACE=files
           export ZQD_DATA_URI=s3://zq-ci-test/$(date +%y-%m-%d).${{ github.run_number }}
           export ZQD_ECR_HOST=${{ secrets.AWS_ECR_HOST }}
+          make install
+          mkdir testlog         
+          aws s3 rm --recursive $ZQD_DATA_URI || : # clear data from previous run
           aws eks --region us-east-2 update-kubeconfig --name zq-test
+          #
+          # Create temporary namespace for tests and deploy zsvc
+          #     
           kubectl create ns $NAMESPACE || :
           kubectl config set-context --current --namespace=$NAMESPACE
           helm uninstall zsrv || :
-          k8s/postgres-secret.sh
+          k8s/postgres-secret.sh || :
           make helm-install PG_PERSIST=false
-          kubectl get pod
           kubectl wait --for=condition=Ready --timeout=60s pods/zsrv-postgresql-0
-  eks-test-zapi-1:
-    # Run smoke tests for zqd image prior to deploying to ci-master.
-    needs: eks-test-deploy
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/setup-go@v2
-        with:
-          go-version: '1.15'
-      - uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-region: us-east-2
-      - uses: actions/checkout@v1
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-      - run: |
-          set -x
-          SPACE=files
-          ZQD_DATA_URI=s3://zq-ci-test/$(date +%y-%m-%d).${{ github.run_number }}
-          mkdir testlog
-          make install
-          aws eks --region us-east-2 update-kubeconfig --name zq-test
-          kubectl config set-context --current --namespace=ci-${{ github.run_number }}
+          kubectl wait --for=condition=available --timeout=120s deployment/zsrv-recruiter
+          kubectl wait --for=condition=available --timeout=120s deployment/zsrv-root
+          kubectl wait --for=condition=available --timeout=120s deployment/zsrv-worker
+          #
+          # Scale cluster for tests and set up port-forwards,
+          # then wait for the forwarded ports to be available.
+          #
           kubectl scale --replicas=4 deployment/zsrv-worker
           kubectl port-forward svc/zsrv-recruiter 8020:9867 &
           kubectl port-forward svc/zsrv-root 9867:9867 &
-          sleep 2 # wait for port-forwards to complete
+          for i in {1..5} ; do
```

Nit: We usually don't include a space before a semicolon in sh scripts.
```suggestion
          for i in {1..5}; do
```